src/main/scala/io/citrine/lolo/linear/GuessTheMean.scala

```diff
@@ -18,7 +20,7 @@ case class GuessTheMeanLearner() extends Learner {
     val data = trainingData.map(_._2).zip(weights.getOrElse(Seq.fill(trainingData.size)(1.0)))
     val mean = data.head._1 match {
       case x: Double => data.asInstanceOf[Seq[(Double, Double)]].map(p => p._1 * p._2).sum / data.map(_._2).sum
-      case x: Any => data.groupBy(_._1).mapValues(_.map(_._2).sum).maxBy(_._2)._1
+      case x: Any => Random.shuffle(data.groupBy(_._1).mapValues(_.map(_._2).sum)).maxBy(_._2)._1
```

Comment:
```suggestion
      case x: Any => Random.shuffle(data.groupBy(_._1).mapValues(_.map(_._2).sum).toSeq).maxBy(_._2)._1
```

I'm a bit uncomfortable relying on the stability of the order of a `MapValues`.  `toSeq` makes the order a bit more explicit.